micro_dl/light/data.py

```diff
@@ -109,50 +110,90 @@ def __init__(
         split_ratio: float,
         batch_size: int = 16,
         num_workers: int = 8,
+        architecture: Literal["2.5D", "2D", "3D"] = "2.5D",
         yx_patch_size: tuple[int, int] = (256, 256),
         augment: bool = True,
+        caching: bool = False,
     ):
         super().__init__()
         self.data_path = data_path
         self.source_channel = source_channel
         self.target_channel = target_channel
         self.batch_size = batch_size
         self.num_workers = num_workers
+        self.target_2d = True if architecture == "2.5D" else False
         self.z_window_size = z_window_size
         self.split_ratio = split_ratio
         self.yx_patch_size = yx_patch_size
         self.augment = augment
+        self.caching = caching
+
+    def _cache(self, lazy_plate: Plate) -> Plate:
+        self.tmp_zarr = os.path.join(
+            tempfile.gettempdir(), os.path.basename(self.data_path)
+        )
+        logging.info(f"Caching dataset at {self.tmp_zarr}.")
```

Comment:
I didn't catch this info on the terminal. It will be useful to point the logger to the same directory as used by `lightning_logs`, i.e., `default_root_dir/lightning_logs/yyyymmdd-hhmmss`.